test/python/visualization/test_dag_drawer.py

```diff
@@ -36,6 +37,32 @@ def test_dag_drawer_invalid_style(self):
         """Test dag draw with invalid style."""
         self.assertRaises(VisualizationError, dag_drawer, self.dag, style="multicolor")
 
+    def test_dag_drawer_checks_filename_correct_format(self):
+        """filename must contain name and extension"""
+        try:
+            dag_drawer(self.dag, filename="aaabc")
+            self.fail("Expected error not raised!")
+        except InvalidFileError as exception_instance:
+            self.assertEqual(
+                exception_instance.message,
+                "Parameter 'filename' must be in format 'name.extension'",
+            )
+
+    def test_dag_drawer_checks_filename_extension(self):
+        """filename must have a valid extension"""
+        try:
+            dag_drawer(self.dag, filename="aa.abc")
+            self.fail("Expected error not raised!")
+        except InvalidFileError as exception_instance:
+            self.assertEqual(
+                exception_instance.message,
+                "Filename extension must be one of: bmp canon cgimage cmap cmapx cmapx_np "
+                "dot dot_json eps exr fig gd gd2 gif gv icns ico imap imap_np ismap jp2 "
+                "jpe jpeg jpg json json0 mp pct pdf pic pict plain plain-ext png pov "
+                "ps ps2 psd sgi svg svgz tga tif tiff tk vdx vml vmlz vrml wbmp webp "
+                "xdot xdot1.2 xdot1.4 xdot_json",
```

Comment:
We might not want to test that exactly this set of extensions are supported - it'll be ok just to test that the message starts with "Filename extension must be one of:"